src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -74,12 +80,12 @@ private void OnSolutionOpenedOrClosed(object sender, EventArgs e)
 
         public bool DeleteSection(string section)
         {
-            return SolutionSettings.DeleteSection(section);
+            return CanChangeSettings && SolutionSettings.DeleteSection(section);
```

Comment:
Yes, when there is an exception, user will get message, but if there is no exception and LoadDefaultSetting returns null, we will show nothing. For now, I think it's impossible to return null without exception, but it might be in future, or we can add some comments in LoadDefaultSetting to remind us to never return null if there is no exception.